Travel Scheduler/View Controllers/FirstScreenViewController.m

```diff
@@ -143,46 +211,76 @@ - (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event
 
 #pragma mark - UISearchBar delegate method
 
-- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText
+{
+    [self.autocompleteTableView reloadData];
     if (searchText.length != 0) {
-        //TODO(Franklin): place API stuff like autocomplete here
-        self.userSpecifiedPlaceToVisit = searchText;
+        [self.autocompleteTableView reloadData];
+        [_fetcher sourceTextHasChanged:searchText];
+        [self.view addSubview:self.autocompleteTableView];
+        [self.autocompleteTableView reloadData];
```

Comment:
Why have so many calls to this instead of doing all your work and then having one call to reload at the end of the method? Maybe I'm missing something, so feel free to clarify!

Reply:
This is mainly due to the fact that when the user wants to change the location and begins to edit the search bar we want the table view to be up to date with what the user has already entered and the other reload data is for the first instance when the user begins to type and table view is created.

This is mainly because of the animations and how we created them to slide in once the user touched the search bar.

Reply:
Honestly, After rethinking through the process the reloadData method call should not be there since I made the first call on line 216. I have removed that.